Master

notnami opened the pull request:
The config parsing routine was moved to a new separate module, called helpers.py (I considered adding it to utils.py, but that module seems dedicated to unit conversions and other mathematical operations). 

This move makes the config parsing function available independently from the command line script. This can make it more convenient to build applications using this package, as login credentials may be read from an already existing configuration file (or stored there for future use). For example:

```
from nest import Nest, get_auth_credentials

username, password = get_auth_credentials(…)

if username and password:
    nest = Nest(username, password)
else:
    # request credentials
    …
```

Potentially, this module can be integrated into the existing Nest class in the nest.py module, where the username and password arguments can be made optional. However, I didn’t want to change the existing structure of nest.py at this stage in case there are objections to my proposed reorganisation.

jkoelker commented:
Thanks! I like having it in `helpers.py`, good call. As far as changing the `Nest` class, I'd rather not make it read the config file, but It would be cool to have a small helper (also in `helpers.py` that would wrap the creation of the `Nest` object):

```python
import contextlib
from . import nest

@contextlib.contextmanager
def nest_login(config_path=None):
    username, password = get_auth_credentials(config_path):
    yield nest.Nest(username, password)
```

I'm not 100% on using a context manager here since it really is just the same as `return Nest(...)`, But it follows the pattern with using the `Nest` class directly as a context manager. Lemme know what you think, and I'll get this merged in and cut the 2.9.3 release.

notnami commented:
Thanks for the feedback.

I have added a context manager. It's slightly more complicated than what you had suggested because I added the option of passing the ```username``` and ```password``` parameters explicitly. I realise that this somewhat duplicates the behaviour of the ```Nest``` class, but I felt that otherwise the API might be too opaque. 

I have also added an exception for when the credentials are not found in the configuration file and not provided explicitly.

jkoelker commented:
Awesome thanks, I'll push the release up to pypi shortly.
